src/include/sof/audio/module_adapter/library/system_agent.h

```diff
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifndef _SYSTEM_AGENT_H
+#define _SYSTEM_AGENT_H
```

Comment:
Is this for generic agent or IADK C++ agent ? Need to be careful for partitioning.